backend/src/services/enhanced_weather_service.py

```
@@ -76,19 +76,25 @@ async def initialize(self) -> None:
             
         except Exception as e:
             logger.error(f"Failed to initialize enhanced weather service: {e}")
-            raise ConfigurationError(f"Enhanced weather service initialization failed: {e}")
+            raise ConfigurationError(
+                f"Enhanced weather service initialization failed: {e}"
+            )
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)